### PR TITLE
ananicy-cpp: patch to fix build on llvm_19

### LINKED ISSUE #367925: Build failure: ananicy-cpp
## Steps To Reproduce

1. `nix-build -A ananicy-cpp` on master (f8aa427967411ed1843040134f742d68fa400f73)

## Build log

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/nmmxl1ai13j7nhyf1khzyakv66j5yj8g-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/x0ag56d2hv9c362z15sq2rxarjq5dz6y-allow-regex-pattern-matching.patch
patching file .gitlab-ci.yml
patching file CMakeLists.txt
patching file cmake/StandardProjectSettings.cmake
patching file include/core/rules.hpp
patching file install-deps.sh
patching file src/rules.cpp
patching file src/tests/CMakeLists.txt
patching file src/tests/unit-utility.cpp
patching file include/utility/regex_utils.hpp
patching file src/regex_utils.cpp
applying patch /nix/store/ygnzmpz15j185n09r6gmc9hfd1c5dmb6-match-wrappers.patch
patching file src/worker.cpp
applying patch /nix/store/1zcm1njizcg2fkc14bfp1wnfqf1qqi5x-reliable-mounts-file.patch
patching file src/platform/linux/cgroups.cpp
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/share/doc/ananicy-cpp -DCMAKE_INSTALL_INFODIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/xl0vlc2wdchfbq8536zs19pj2r3xdmma-clang-wrapper-19.1.5/bin/strip -DCMAKE_RANLIB=/nix/store/xl0vlc2wdchfbq8536zs19pj2r3xdmma-clang-wrapper-19.1.5/bin/ranlib -DCMAKE_AR=/nix/store/xl0vlc2wdchfbq8536zs19pj2r3xdmma-clang-wrapper-19.1.5/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/bkzm3ab38iamay6abm9d4dxlarxg9ryv-ananicy-cpp-1.1.1 -DBPF_BUILD_LIBBPF:BOOL=FALSE -DENABLE_REGEX_SUPPORT:BOOL=TRUE -DENABLE_SYSTEMD:BOOL=TRUE -DUSE_BPF_PROC_IMPL:BOOL=TRUE -DUSE_EXTERNAL_FMTLIB:BOOL=TRUE -DUSE_EXTERNAL_JSON:BOOL=TRUE -DUSE_EXTERNAL_SPDLOG:BOOL=TRUE -DVERSION:STRING=1.1.1
Version: 1.1.1
-- version for cmake: 1.1.1
-- The CXX compiler identification is Clang 19.1.5
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/xl0vlc2wdchfbq8536zs19pj2r3xdmma-clang-wrapper-19.1.5/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Checking compiler flags support
-- Performing Test CXX_ACCEPTS_G3
-- Performing Test CXX_ACCEPTS_G3 - Success
-- Performing Test CXX_ACCEPTS_G
-- Performing Test CXX_ACCEPTS_G - Success
-- Performing Test CXX_ACCEPTS_OG
-- Performing Test CXX_ACCEPTS_OG - Success
-- Performing Test CXX_ACCEPTS_MARCH_NATIVE
-- Performing Test CXX_ACCEPTS_MARCH_NATIVE - Success
-- Performing Test CXX_ACCEPTS_THIN_LTO
-- Performing Test CXX_ACCEPTS_THIN_LTO - Failed
-- Performing Test CXX_ACCEPTS_WHPR_VTABLE
-- Performing Test CXX_ACCEPTS_WHPR_VTABLE - Failed
-- Performing Test CXX_ACCEPTS_STRICT_VTABLE
-- Performing Test CXX_ACCEPTS_STRICT_VTABLE - Success
-- [ccache] Disabled.
-- [cppcheck] Disabled.
-- [clang-tidy] Disabled.
-- [iwyu] Disabled.
CMake Warning at cmake/CPM.cmake:61 (message):
  Your project is using an unstable development version of CPM.cmake.  Please
  update to a recent release if possible.  See
  https://github.com/cpm-cmake/CPM.cmake for details.
Call Stack (most recent call first):
  CMakeLists.txt:18 (include)


-- Linux system detected, adding specific sources files
-- Configuring threads
-- [threads] Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- [threads] Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- [threads] Found Threads: TRUE
-- Configuring nlohmann_json
-- [nlohmann_json] Found nlohmann_json: /nix/store/75a52g5kjbx98i6srxvl3prbpkhndjci-nlohmann_json-3.11.3/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.11.3", minimum required is "3.9")
-- [fmtlib] Configuring fmtlib
[fmtlib] Using external fmtlib
-- [pcre2] Configuring pcre2
-- [pcre2] Found PkgConfig: /nix/store/0d4m43yp69lrm8imxbqgl9zxjwwz52jw-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- [pcre2] Checking for module 'libpcre2-8'
-- [pcre2]   Found libpcre2-8, version 10.44
-- [spdlog] Configuring spdlog
[spdlog] Using external spdlog
-- Configuring systemd
-- The C compiler identification is Clang 19.1.5
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/xl0vlc2wdchfbq8536zs19pj2r3xdmma-clang-wrapper-19.1.5/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found clang version: 19.1.5
-- Found LibBpf: /nix/store/7x2jlhj9wjjg4b5azjyphkl853c142ri-libbpf-1.4.7/lib/libbpf.so
-- Found BpfObject: /nix/store/19rv2vl9dnxyxa9r76k7kj3jhkfr6bml-bpftools-6.12/bin/bpftool
-- BPF system include flags: -idirafter /nix/store/nd72mapx3gzc789k06gyz2wl20q3jj8m-compiler-rt-libc-19.1.5-dev/include -idirafter /nix/store/jwrmbs5kcc92wv824xhxpypcj1iihl83-pcre2-10.44-dev/include -idirafter /nix/store/f9b0zf1ff2h2jhvcddllp1hjwhnfsyi7-spdlog-1.14.1-dev/include -idirafter /nix/store/bidr7f7bayhc3ixfpckrl43rmlkkf4vx-fmt-10.2.1-dev/include -idirafter /nix/store/75a52g5kjbx98i6srxvl3prbpkhndjci-nlohmann_json-3.11.3/include -idirafter /nix/store/ixwyfddsia31qxchy5rjm3nz0j0f4011-systemd-256.8-dev/include -idirafter /nix/store/1lggwqzapn5mn49l9zy4h566ysv9kzdb-zlib-1.3.1-dev/include -idirafter /nix/store/7x2jlhj9wjjg4b5azjyphkl853c142ri-libbpf-1.4.7/include -idirafter /nix/store/wk9dg3asb56f6kdi6v6hxrh03shick2d-elfutils-0.192-dev/include -idirafter /nix/store/xl0vlc2wdchfbq8536zs19pj2r3xdmma-clang-wrapper-19.1.5/resource-root/include -idirafter /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include
-- BPF target arch: x86
-- Configuring done (3.6s)
-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
[  4%] [clang] Building BPF object: ananicy_cpp
clang: [0;1;35mwarning: argument unused during compilation: '--gcc-toolchain=/nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116' [-Wunused-command-line-argument]
[  8%] [skel]  Building BPF skeleton: ananicy_cpp
[ 12%] Building C object libananicycpp_bpf/CMakeFiles/ananicy_cpp_bpf_c.dir/src/bpf_program_utils.c.o
[ 20%] Building C object libananicycpp_bpf/CMakeFiles/ananicy_cpp_bpf_c.dir/src/btf_helpers.c.o
[ 20%] Building C object libananicycpp_bpf/CMakeFiles/ananicy_cpp_bpf_c.dir/src/uprobe_helpers.c.o
[ 25%] Building C object libananicycpp_bpf/CMakeFiles/ananicy_cpp_bpf_c.dir/src/trace_helpers.c.o
[ 29%] Linking C static library libananicy_cpp_bpf_c.a
[ 29%] Built target ananicy_cpp_bpf_c
[ 33%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/main.cpp.o
[ 37%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/rules.cpp.o
[ 41%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/utility/argument_parsing/argument_parser.cpp.o
[ 45%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/utils.cpp.o
[ 50%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/config.cpp.o
[ 54%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/worker.cpp.o
[ 58%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/regex_utils.cpp.o
[ 62%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/utility/argument_parsing/argument.cpp.o
[ 66%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/linux/process.cpp.o
[ 70%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/linux/backtrace.cpp.o
[ 75%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/linux/priority.cpp.o
[ 79%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/linux/process_info.cpp.o
[ 83%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/linux/cgroups.cpp.o
[ 87%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/systemd/service.cpp.o
[ 91%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/linux/debug.cpp.o
[ 95%] Building CXX object CMakeFiles/ananicy-cpp.dir/src/platform/linux/singleton_process.cpp.o
In file included from /build/source/src/platform/linux/process.cpp:5:
In file included from /build/source/include/core/process.hpp:13:
In file included from /build/source/include/utility/synchronized_queue.hpp:15:
/build/source/include/utility/atomic_queue/atomic_queue.h:396:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  396 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:402:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  402 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
/build/source/include/utility/atomic_queue/atomic_queue.h:524:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  524 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:530:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  530 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
In file included from /build/source/src/platform/linux/process_info.cpp:2:
In file included from /build/source/include/core/process.hpp:13:
In file included from /build/source/include/utility/synchronized_queue.hpp:15:
/build/source/include/utility/atomic_queue/atomic_queue.h:396:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  396 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:402:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  402 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
/build/source/include/utility/atomic_queue/atomic_queue.h:524:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  524 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:530:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  530 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
In file included from /build/source/src/main.cpp:7:
In file included from /build/source/include/core/process.hpp:13:
In file included from /build/source/include/utility/synchronized_queue.hpp:15:
/build/source/include/utility/atomic_queue/atomic_queue.h:396:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  396 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:402:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  402 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
/build/source/include/utility/atomic_queue/atomic_queue.h:524:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  524 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:530:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  530 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
In file included from /build/source/src/worker.cpp:5:
In file included from /build/source/include/core/worker.hpp:9:
In file included from /build/source/include/core/process.hpp:13:
In file included from /build/source/include/utility/synchronized_queue.hpp:15:
/build/source/include/utility/atomic_queue/atomic_queue.h:396:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  396 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:402:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  402 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
/build/source/include/utility/atomic_queue/atomic_queue.h:524:31: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  524 |         return Base::template do_pop_any(states_[index], elements_[index]);
      | [0;1;32m                              ^
/build/source/include/utility/atomic_queue/atomic_queue.h:530:24: [0;1;31merror: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  530 |         Base::template do_push_any(std::forward<U>(element), states_[index], elements_[index]);
      | [0;1;32m                       ^
4 errors generated.
make[2]: *** [CMakeFiles/ananicy-cpp.dir/build.make:202: CMakeFiles/ananicy-cpp.dir/src/platform/linux/process.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
4 errors generated.
make[2]: *** [CMakeFiles/ananicy-cpp.dir/build.make:118: CMakeFiles/ananicy-cpp.dir/src/worker.cpp.o] Error 1
4 errors generated.
make[2]: *** [CMakeFiles/ananicy-cpp.dir/build.make:76: CMakeFiles/ananicy-cpp.dir/src/main.cpp.o] Error 1
4 errors generated.
make[2]: *** [CMakeFiles/ananicy-cpp.dir/build.make:272: CMakeFiles/ananicy-cpp.dir/src/platform/linux/process_info.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/ananicy-cpp.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

</details>

## Additional context

Also, one of the dependencies fails: `schedtool`

## Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.2-zen1, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.92.0-dev-pre20241211-92ed9fe
System type: x86_64-linux
Additional system types: aarch64-linux, i686-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/perchun/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/perchun/.nix-profile/etc/xdg/nix/nix.conf:/home/perchun/.local/state/nix/profile/etc/xdg/nix/nix.conf:/home/perchun/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/perchun/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/hd8gdx3wk23c3vdql1091bbm7wcrhfhy-lix-2.92.0-dev-pre20241211-92ed9fe/share`
 - nixpkgs: `/etc/nix/path/nixpkgs`
```

## Notify maintainers

@abbradar @artturin @johnrtitor @diniamo

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by JohnRTitor**
Due to changes in template in llvm_19.

> error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
Link: https://github.com/llvm/llvm-project/issues/94049

Closes https://github.com/NixOS/nixpkgs/issues/367925#issuecomment-2561175494

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JohnRTitor** [comment]
@ofborg build ananicy-cpp

**Review by paparodeo — CHANGES_REQUESTED** [review]
why is this a regression in clang? the links you posted were to fixes in flang to fix the newly reported error in clang-19.

[edit] flang is a fortran compiler I believe.

**Comment by JohnRTitor** [comment]
> why is this a regression in clang? the links you posted were to fixes in flang to fix the newly reported error in clang-19.

Sorry, seem to mixed up the things. But anyway I'll apply the patch, and see if it builds later (not near a computer right now)

**Comment by paparodeo** [comment]
> > why is this a regression in clang? the links you posted were to fixes in flang to fix the newly reported error in clang-19.
> 
> Sorry, seem to mixed up the things. But anyway I'll apply the patch, and see if it builds later (not near a computer right now)

I opened an issue too: https://gitlab.com/ananicy-cpp/ananicy-cpp/-/issues/79

**Comment by JohnRTitor** [comment]
The above patch indeed fixes the build. Perhaps you could open a PR upstream?

Also pinging @aviallon, an ananicy-cpp maintainer

**Comment by JohnRTitor** [comment]
@ofborg build ananicy-cpp

File: pkgs/by-name/an/ananicy-cpp/package.nix [hunk 1]
```diff
@@ -38,6 +38,7 @@ clangStdenv.mkDerivation (finalAttrs: {
     ./match-wrappers.patch
     # https://gitlab.com/ananicy-cpp/ananicy-cpp/-/merge_requests/27
     ./reliable-mounts-file.patch
+    ./fix-llvm-19-build.patch
```
Comment (paparodeo) [inline pkgs/by-name/an/ananicy-cpp/package.nix:41]:
```suggestion
    # https://gitlab.com/ananicy-cpp/ananicy-cpp/-/issues/79
    ./fix-llvm-19-build.patch
```

File: pkgs/by-name/an/ananicy-cpp/package.nix [hunk 2]
```diff
@@ -16,7 +16,7 @@
   withBpf ? true,
 }:
 
-clangStdenv.mkDerivation (finalAttrs: {
+llvmPackages.stdenv.mkDerivation (finalAttrs: {
```
Comment (paparodeo) [inline pkgs/by-name/an/ananicy-cpp/package.nix:19]: what's wrong with leaving it as `clangStdenv` ?

**Comment by aviallon** [comment]
> The above patch indeed fixes the build. Perhaps you could open a PR upstream?
> 
> Also pinging @aviallon, an ananicy-cpp maintainer

Thanks for pinging me. This part of the code was written by @vnepogodin, but it seems trivial enough for me to upstream.

File: pkgs/by-name/an/ananicy-cpp/package.nix (hunk 2 above)
Comment (JohnRTitor) [reply pkgs/by-name/an/ananicy-cpp/package.nix:19]: Nothing, for some reason I thought `clangStdenv` was an alias of `llvmPackages.stdenv`.

**Comment by vnepogodin** [comment]
> > The above patch indeed fixes the build. Perhaps you could open a PR upstream?
> > 
> > Also pinging @aviallon, an ananicy-cpp maintainer
> 
> Thanks for pinging me. This part of the code was written by @vnepogodin, but it seems trivial enough for me to upstream. 

Well, we might even queue a new release(at least minor one)

File: pkgs/by-name/an/ananicy-cpp/package.nix (hunk 1 above)
Comment (JohnRTitor) [reply pkgs/by-name/an/ananicy-cpp/package.nix]: Linked to the upstream MR instead

**Comment by JohnRTitor** [comment]
Opened upstream MR and cherry-picked the commit here.

https://gitlab.com/ananicy-cpp/ananicy-cpp/-/merge_requests/30

**Review by paparodeo — APPROVED** [review]
LGTM build on x64 linux

**Pull request merged**